realm/realm-library/src/androidTest/java/io/realm/BulkInsertTests.java

```diff
@@ -0,0 +1,491 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+
+import io.realm.entities.AllTypes;
+import io.realm.entities.AllTypesPrimaryKey;
+import io.realm.entities.CyclicType;
+import io.realm.entities.Dog;
+import io.realm.entities.DogPrimaryKey;
+import io.realm.entities.NoPrimaryKeyWithPrimaryKeyObjectRelation;
+import io.realm.entities.NullTypes;
+import io.realm.entities.PrimaryKeyAsBoxedShort;
+import io.realm.entities.PrimaryKeyAsLong;
+import io.realm.entities.PrimaryKeyAsString;
+import io.realm.entities.PrimaryKeyWithNoPrimaryKeyObjectRelation;
+import io.realm.entities.pojo.AllTypesRealmModel;
+import io.realm.entities.pojo.InvalidRealmModel;
+import io.realm.exceptions.RealmException;
+import io.realm.rule.TestRealmConfigurationFactory;
+
+import static io.realm.internal.test.ExtraTests.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+public class BulkInsertTests {
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
+
+    private Realm realm;
+    private RealmConfiguration realmConfig;
+
+    @Before
+    public void setUp() {
+        realmConfig = configFactory.createConfiguration();
+        realm = Realm.getInstance(realmConfig);
+    }
+
+    @After
+    public void tearDown() {
+        if (realm != null) {
+            realm.close();
+        }
+    }
+
+    @Test
+    public void insertToRealm() {
+        Date date = new Date();
+        Dog dog = new Dog();
+        dog.setName("Fido");
+        RealmList<Dog> list = new RealmList<Dog>();
+        list.add(dog);
+
+        AllTypes allTypes = new AllTypes();
+        allTypes.setColumnString("String");
+        allTypes.setColumnLong(1l);
+        allTypes.setColumnFloat(1f);
+        allTypes.setColumnDouble(1d);
+        allTypes.setColumnBoolean(true);
+        allTypes.setColumnDate(date);
+        allTypes.setColumnBinary(new byte[]{1, 2, 3});
+        allTypes.setColumnRealmObject(dog);
+        allTypes.setColumnRealmList(list);
+
+        realm.beginTransaction();
+        realm.insertToRealm(allTypes);
+        realm.commitTransaction();
+
+        AllTypes realmTypes = realm.where(AllTypes.class).findFirst();
```

Comment:
Perhaps use `AllJavaTypes` instead since right now there is some code paths in the annotation processor not getting touched (int/short)